core/replacement_test.go

```diff
@@ -180,3 +171,66 @@ func TestErrorBadTemplateExec(t *testing.T) {
 	err := rep.Execute(ReplaceData{})
 	then.NotNil(t, err)
 }
+
+func testGlobs(t *testing.T, paths []string, rep Replacement) {
+	toReplace := `{
+  "version": "1.0.0"
+}`
+	replaceWith := "1.2.3"
+
+	for _, path := range paths {
+		split := strings.Split(path, "/")
+		if split[0] != path {
+			err := os.Mkdir(split[0], os.ModePerm)
+			then.Nil(t, err)
+		}
+		err := os.WriteFile(path, []byte(toReplace), os.ModePerm)
+		then.Nil(t, err)
+	}
+
+	err := rep.Execute(ReplaceData{
+		VersionNoPrefix: replaceWith,
+	})
+	then.Nil(t, err)
+
+	expected := `{
+  "version": "` + replaceWith + `"
+}`
+
+	for _, path := range paths {
+		then.FileContents(t, expected, path)
+	}
+}
+
+func TestGlobs(t *testing.T) {
```

Comment:
just going to rewrite this test in my style, I think you hit all the cases but I think I have some util methods already in place to make this a little easier.